python_scripts/metrics_regression.py

```diff
@@ -185,21 +206,62 @@
 model_transformed_target.fit(data_train, target_train)
 target_predicted = model_transformed_target.predict(data_test)
 
-# %%
-predicted_actual = {
-    "True values (k$)": target_test, "Predicted values (k$)": target_predicted}
-predicted_actual = pd.DataFrame(predicted_actual)
+fig, axs = plt.subplots(ncols=2, figsize=(13, 5))
+
+PredictionErrorDisplay.from_predictions(
+    y_true=target_test,
+    y_pred=target_predicted,
+    kind="actual_vs_predicted",
+    scatter_kwargs={"alpha": 0.5},
+    ax=axs[0]
+)
+axs[0].axis("square")
+axs[0].set_xlabel("Predicted values (k$)")
+axs[0].set_ylabel("True values (k$)")
+
+PredictionErrorDisplay.from_predictions(
+    y_true=target_test,
+    y_pred=target_predicted,
+    kind="residual_vs_predicted",
+    scatter_kwargs={"alpha": 0.5},
+    ax=axs[1]
+)
+axs[1].axis("square")
+axs[1].set_xlabel("Predicted values (k$)")
+axs[1].set_ylabel("Residual values (k$)")
+
+_ = fig.suptitle("Regression using a model that\ntransforms the target "
+                 "before fitting", y=1.1)
+
+# %% [markdown]
+# The model with the transformed target seems to exhibit fewer structure in its
+# residuals: over-estimation and under-estimation errors seems to be more
+# balanced.
+#
+# We can confirm this by computing the previously mentioned metrics and observe
+# that they all improved w.r.t. the linear regression model without the target
+# transformation.
 
 # %%
-sns.scatterplot(data=predicted_actual,
-                x="True values (k$)", y="Predicted values (k$)",
-                color="black", alpha=0.5)
-plt.axline((0, 0), slope=1, label="Perfect fit")
-plt.axis('square')
-plt.legend()
-_ = plt.title("Regression using a model that\n transform the target before "
-              "fitting")
+print(f"Mean absolute error: "
+      f"{mean_absolute_error(target_test, target_predicted):.3f} k$")
+
+print(f"Median absolute error: "
+      f"{median_absolute_error(target_test, target_predicted):.3f} k$")
+
+print(f"Mean absolute percentage error: "
+      f"{mean_absolute_percentage_error(target_test, target_predicted) * 100:.3f} %")
 
 # %% [markdown]
-# Thus, once we transformed the target, we see that we corrected some of the
-# high values.
+# While a common practice, performing such a target transformation for linear
+# regression is often disapproved by statisticians. It is mathematically more
+# justified to instead adapt the loss function of the regression model itself,
+# for instance by fitting a `PoissonRegressor` or a `TweedieRegressor` model
+# instead of `LinearRegression`. In particular those models indeed use an
+# internal "log link" function that makes them more suited for this kind of
+# positive-only target data distributions, but this analysis is beyond the scope
+# of this MOOC.
+#
+# The interested readers are encouraged to learn more about those models, in
+# particular by reading the user guide and the examples linked for their API
+# documentation pages above.
```

Comment:
Thanks for integrating my suggestion. However we need to make the class names for `PoissonRegressor` and `TweedieRegressor` linked to the scikit-learn API documentation for each class.

Or alternatively we should change this last paragraph to link to the user guide on generalized linear models.

Reply:
I chose the former as the user guide on generalized linear models contains some other models (including the logistic regression) and may bring more questions than answers.